client/starwhale/api/_impl/dataset/loader.py

```diff
@@ -120,17 +119,12 @@ def __init__(
             raise ValueError(f"cache_size({cache_size}) must be a positive int number")
         self._cache_size = cache_size
 
-    def _load_dataset_auth_env(self) -> None:
-        # TODO: support multi datasets
-        if self.dataset_uri.instance_type == InstanceType.STANDALONE:
-            auth_env_fpath = (
-                DatasetStorage(self.dataset_uri).snapshot_workdir / AUTH_ENV_FNAME
-            )
-            load_dotenv(auth_env_fpath)
-
     def _get_store(self, row: TabularDatasetRow) -> ObjectStore:
         with self._store_lock:
-            _k = f"{self.dataset_uri}.{row.data_link.scheme}.{row.auth_name}"
+            _up = urlparse(row.data_link.uri)
+            parts = _up.path.lstrip("/").split("/", 1)
+            c_k = row.data_link.uri.replace(parts[-1], "")
```

Comment:
use another meaningful variable name(`c_k`)? 
In addition, there both are tmp/private variable(`_up`, `_k`) and normal variable(`parts`, `c_k`), use a uniform pattern?

Reply:
If `parts[-1]` content is also a part as row.data_link.uri netloc, the replace operation will cause the unexpected result.
for example:
the URI is `http://www.test.com/a/test`, and the c_k is `http://www..com/a/`.

In addition, the query part should also be truncated.

Reply:
```
_up = urlparse(row.data_link.uri)
parts = _up.path.lstrip("/").split("/", 1)
c_k = row.data_link.uri.replace(parts[-1], "")
_k = f"{self.dataset_uri}.{c_k}"
```
-->
```
_up = urlparse(row.data_link.uri)
_parts = _up.path.lstrip("/").split("/", 1)
_k = f"{self.dataset_uri}.{_up.netloc}.{parts[0]}
```
?

Reply:
> If `parts[-1]` content is also a part as row.data_link.uri netloc, the replace operation will cause the unexpected result. for example: the URI is `http://www.test.com/a/test`, and the c_k is `http://www..com/a/`.
> 
> In addition, the query part should also be truncated.

Eventhough the cache key is wired, it works. Yes, the query part need to be truncated

Reply:
> ```
> _up = urlparse(row.data_link.uri)
> parts = _up.path.lstrip("/").split("/", 1)
> c_k = row.data_link.uri.replace(parts[-1], "")
> _k = f"{self.dataset_uri}.{c_k}"
> ```
> 
> -->
> 
> ```
> _up = urlparse(row.data_link.uri)
> _parts = _up.path.lstrip("/").split("/", 1)
> _k = f"{self.dataset_uri}.{_up.netloc}.{parts[0]}
> ```
> 
> ?

A problem with `parts[0]` is that URI which has only one path part will have a unique cache key itself. The cache mechnism fails when uri looks like `data_ubyte_0.swds_bin`. If `parts[-1]` is replace with empty string, the cache keys reduce to `{self.dataset_uri}.` and the cache works.

Reply:
> use another meaningful variable name(`c_k`)? In addition, there both are tmp/private variable(`_up`, `_k`) and normal variable(`parts`, `c_k`), use a uniform pattern?

done